src/core/include/openvino/core/logging.hpp

```diff
@@ -0,0 +1,21 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <functional>
+#include <string_view>
+
+#include "openvino/core/core_visibility.hpp"
+
+namespace ov {
+
+/**
+ * @brief Sets and resets log messages handling callback.
+ * @param [in] callback     A pointer to callable object which is called on single message logging. Empty object is fine
+ * (no messages output). `nullptr' resets internal callback to its default (messages are streamed to `std::cout').
```

Comment:
To consider:
Consider to avoid pass null pointer always pass some instance of callback
For reset add dedicated function like in dev_api or:
- `set_log_callback()` -> no user object set default.
- `set_log_callback(ov::DEFAULT)` -> use tag to indicate default.